In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split

In [30]:
media_df = pd.read_csv('https://raw.githubusercontent.com/shivang98/Social-Network-ads-Boost/master/Social_Network_Ads.csv',index_col=0)

In [31]:
X = media_df.iloc[:,0:-1]
y = media_df.iloc[:,-1]
# split training and testing
X_train, X_test, y_train, y_test = train_test_split(X,
                                                    y,
                                                    test_size=0.2, random_state = 123)

In [32]:
media_df_train = X_train
media_df_train = pd.merge(media_df_train, y_train, left_index=True, right_index=True)
media_df_train.head()

Gender  Age  EstimatedSalary  Purchased
User ID                                          
15584114    Male   24            23000          0
15795298  Female   48           134000          1
15598044  Female   27            84000          0
15692819  Female   57            26000          1
15697020  Female   39            61000          0

In [9]:
media_df_train.isna().any()

Gender             False
Age                False
EstimatedSalary    False
Purchased          False
dtype: bool

We can see that there is no missing values in our dataset.

In [10]:
print(media_df_train.shape)
print(media_df_train.columns)


(320, 4)
Index(['Gender', 'Age', 'EstimatedSalary', 'Purchased'], dtype='object')


It has four columns, one of which is our target: Purchased, a binary variable with 0 being not purchased and 1 being purchased. In the three features, one is Gender which are strings and the other two are floats.

In [11]:
#general summary for all data
media_df_train.describe()

Age  EstimatedSalary   Purchased
count  320.000000       320.000000  320.000000
mean    37.909375     70843.750000    0.375000
std     10.093974     33860.130312    0.484881
min     18.000000     15000.000000    0.000000
25%     30.000000     44000.000000    0.000000
50%     37.000000     71000.000000    0.000000
75%     46.000000     88000.000000    1.000000
max     60.000000    150000.000000    1.000000

We can see that the mean purchased in 0.375 which is different from purchasing by chance of 0.5. This indicates that our research question is meaningful and it is worth investigating how these three factors affect the purchasing behavior.

In [13]:
#Summary for negative class, which is not purhcased
media_df_train[media_df_train['Purchased']==0].describe()

Age  EstimatedSalary  Purchased
count  200.000000       200.000000      200.0
mean    33.180000     61505.000000        0.0
std      7.620892     23782.726196        0.0
min     18.000000     15000.000000        0.0
25%     27.000000     46500.000000        0.0
50%     35.000000     61000.000000        0.0
75%     39.000000     78250.000000        0.0
max     59.000000    141000.000000        0.0

In [14]:
#Summary for negative class, which is not purhcased
media_df_train[media_df_train['Purchased']==1].describe()

Age  EstimatedSalary  Purchased
count  120.000000       120.000000      120.0
mean    45.791667     86408.333333        1.0
std      8.704192     41673.500000        0.0
min     27.000000     20000.000000        1.0
25%     39.000000     40500.000000        1.0
50%     47.000000     90500.000000        1.0
75%     53.000000    122250.000000        1.0
max     60.000000    150000.000000        1.0

Comparing the summary for purchased vs not purchased class, we can see that the mean age and mean estimated salary are higher for purchased class. 